pkg/apply/solver/solver_test.go

```diff
@@ -746,7 +752,12 @@ func TestTaskQueueBuilder_AppendPruneWaitTasks(t *testing.T) {
 					}
 					assert.Equal(t, taskrunner.AllNotFound, actWaitTask.Condition)
 					// Validate the prune wait timeout.
-					assert.Equal(t, tc.options.PruneTimeout, actualTask.(*taskrunner.WaitTask).Timeout)
+					if tc.options.ReconcileTimeout != 0 {
+						assert.Equal(t, tc.options.PruneTimeout, expTsk.Timeout)
+						assert.Equal(t, expTsk.Timeout, actWaitTask.Timeout)
+					} else {
+						assert.Equal(t, tc.options.PruneTimeout, actWaitTask.Timeout)
```

Comment:
This else seems redundant. Isn't this always being tested?

Reply:
No. If `tc.Options.PruneTimeout` is not set, `WaitTask.Timeout` will be zero. We can either test `assert.Equal(t, tc.options.PruneTimeout, actWaitTask.Timeout)` or change the `WaitTask.Timeout` in `tc.expectedTasks` from `1*time.Second` to 0.

Reply:
This can be simplified to the following, right?
```
if tc.options.ReconcileTimeout != 0 {
    assert.Equal(t, tc.options.PruneTimeout, expTsk.Timeout)
}
assert.Equal(t, tc.options.PruneTimeout, actWaitTask.Timeout)
```